File: practice/algorithms/implementation/fair_rations/fair_rations.py
#!/usr/bin/env python

n = int(raw_input().strip())
b = map(int,raw_input().strip().split())

res = 0
for POS in xrange(n - 1):
    if b[POS] % 2 == 1:
        b[POS + 1] += 1
        res += 2
if b[n - 1] % 2 == 0:
    print res
else:
    print "NO"

File: practice/algorithms/implementation/fair_rations/fair_rations.cpp
#include <vector>
#include <iostream>

using namespace std;

int main()
{
    int n, res = 0;
    cin >> n;
    vector<int> b(n);
    for (int pos = 0; pos < n; ++pos)
    {
       cin >> b[pos];
    }
    
    for (int pos = 0; pos < n - 1; ++pos)
    {
        if (b[pos] % 2 == 1)
        {
            b[pos + 1]++;
            res += 2;
        }
    }
    
    if (b[n - 1] % 2 == 0)
    {
        cout << res << endl;
    }
    else
    {
        cout << "NO" << endl;
    }
    
    return 0;
}

